Articles/Blog/SoftwareVariants.md

```diff
@@ -0,0 +1,209 @@
+     - Treat *multioptions* as lists of functionality (e.g., compile multiple variants of CPU and GPU backends that can be selected at runtime). 
+     This approach can still make use of single-source programming patterns and just needs an additional runtime dispatch at a high-level in the program workflow, where latency is usually not an issue. Similar solutions also exist  for runtime-dispatched vectorization control.
+    -- Rationale: avoids incompatibilities, supports feature-complete deployments, and enhances clarity in documentation
+
+3. Avoid modifying the "base" behavior of the software if additional functionality is activated - existing functionality and dependencies, at compile and runtime, stays unchanged.
+    -- Rationale: supports feature-complete deployments and enhances clarity in documentation
+
+4. Add explicit configuration and runtime control to use such opt-in functionality and enhancements.
+    -- Rationale: avoids implicit assumptions that do not necessarily hold true when combined in a combined software ecosystem
+
+### Package managers and deployment
+
+But hey, don't we have package managers to solve this problem for us by keeping track of the right variant?
+Well, in part.
+Of course, modern package managers such as [Spack](https://spack.io) enable developers to define various development environments with exactly chosen and combined variants of software.
+This approach is helpful for correct results.
+Yet, if one needs to use multiple (compile-time) features of a software, this results in a combinatorial explosion of artifacts for development environments and deployments.
+
+On top of that, many popular package managers in use by developers and users are not as powerful as Spack.
+Most do not support binary variants at all -- and are adding them as afterthoughts via package naming extensions, again resulting in combinatorial installs and requiring complex conflict resolutions.
+
+Indeed, independent of whether a package manager supports binary variants well, it is tremendously helpful if a package manager with great dependency resolution can just switch *ON* all options that are *potentially* useful for a system at the same time.
+With that capability, there are fewer modules to build, no environment switching is needed for development, and binary caches can be smaller.
+
+<img src='../../images/Blog_2209_SoftwareVariants_Spack.png' />[Some of the WarpX compile-time options exposed in the Spack package manager.]
+
+### Hands-on examples
+
+**C/C++**
+
+The following design patterns can be used for C/C++ code.
+
+**Header:** Selected opt-in variants should add *extra* header files that expose additional classes and API calls and expose functionality such as MPI-enabled signatures.
+This means at least one public ``facade" header is needed per configuration option, to avoid ``polluting" base functionality with third-party includes.
```

Comment:
Is that the correct formatting in `.md` for double quotes? Looks LaTeX-y :)
```suggestion
This means at least one public "facade" header is needed per configuration option, to avoid "polluting" base functionality with third-party includes.
```